made all widgets scalable (needed for High-DPI-displays)

h-b opened the pull request:
I chose cvui for an image processing project where high-dpi displays may be used (as they are common nowadays). On a display with 180 DPI or more, the widgets look tiny and the text becomes unreadable. Only text() and printf() did support scaling, as they consist only of text. Now all other widgets - including EnhancedWindow - can be scaled analogous to text() and printf(). The internal geometry of all widgets is automatically adapted consistent to their new parameter theFontSize (for example the height of the title bar of a window and dozens of other small things). The size of a widget as defined by the user is not changed.

The samples main-app and ui-enhanced-window-component both now include a cvui::counter which the user can use to change the scaling of the user interface. In my (closed-source) application, instead of a cvui::counter I query the display DPI to define the scaling, so the whole application always has the same physical dimensions on a display, independent from its pixel density.

I think everything works fine. I didn′t care about the Python part, though.

Dovyski commented:
This is outstanding, thank you very much for such a contribution! I am organizing my job schedule for 2020 to accommodate a few hours of work on open source projects every week. I will include cvui on that budged too.

Since this is a significantly large PR, I will need time to review it. Thanks again and I look forward to seeing it incorporated into the lib.

h-b commented:
Thank you for your positive response! I am looking forward to see my commits in your repo. I added two other commits. In the first, I moved EnhancedWindow.h from the corresponding example project to a path next to cvui.h. There are two samples requiring EnhancedWindow.h and for projects who like to use it (like mine), it looks odd to have a dependency on an example.

In the second, I introduced support for colored buttons and counters, because I need colors in my project. With the new argument theInsideColor one can specify the fill color of a button. The other three colors that are used to paint buttons are automatically derived from it, so that they match the original default colors if theInsideColor is the DEFAULT_BUTTON_COLOR 0x424242. I modified the example main-app so it shows one colored button and counter.
